tests/ert_tests/ert3/evaluator/test_evaluator.py

```diff
@@ -159,3 +159,33 @@ def test_evaluator_function(
 
     expected = {iens: {"polynomial_output": data} for iens, data in enumerate(expected)}
     assert expected == evaluation_records
+
+
+@pytest.mark.parametrize(
+    "commandline, parsed_cmd_args",
+    [
+        ("echo inline", ["inline"]),
+        ("bash -c 'echo inline'", ["-c", "echo inline"]),
+        ('bash -c "echo inline"', ["-c", "echo inline"]),
+        ("bash -c echo inline", ["-c", "echo", "inline"]),  # (user error)
```

Comment:
Perhaps what I used it for could serve as an even more complex test case? :)

```suggestion
        ("bash -c echo inline", ["-c", "echo", "inline"]),  # (user error)
        ("sh -c" + `"` + "echo '[0, 1, 2, 3]' > some_numbers.json" + `"`, [ "-c", "echo '[0, 1, 2, 3]' > some_numbers.json"]),
```